test/run_envoy_bazel_coverage.sh

```diff
@@ -31,97 +21,43 @@ else
-NUM_PREVIOUS_GCNO_FILES=0
-for f in $(find -L "${GCNO_ROOT}" -name "*.gcno")
-do
-  rm -f "${f}"
-  let NUM_PREVIOUS_GCNO_FILES=NUM_PREVIOUS_GCNO_FILES+1
-done
-echo "Cleanup completed. ${NUM_PREVIOUS_GCNO_FILES} files deleted."
-
 # Make sure //test/coverage:coverage_tests is up-to-date.
 SCRIPT_DIR="$(realpath "$(dirname "$0")")"
-(BAZEL_BIN="${BAZEL_COVERAGE}" "${SCRIPT_DIR}"/coverage/gen_build.sh ${COVERAGE_TARGETS})
-
-echo "Cleaning .gcda/.gcov from previous coverage runs..."
-NUM_PREVIOUS_GCOV_FILES=0
-for f in $(find -L "${GCOVR_DIR}" -name "*.gcda" -o -name "*.gcov")
-do
-  rm -f "${f}"
-  let NUM_PREVIOUS_GCOV_FILES=NUM_PREVIOUS_GCOV_FILES+1
-done
-echo "Cleanup completed. ${NUM_PREVIOUS_GCOV_FILES} files deleted."
+NO_GCOV=1 "${SCRIPT_DIR}"/coverage/gen_build.sh ${COVERAGE_TARGETS}
 
-# Force dbg for path consistency later, don't include debug code in coverage.
-BAZEL_BUILD_OPTIONS="${BAZEL_BUILD_OPTIONS} -c dbg --copt=-DNDEBUG"
-
-# Run all tests under "bazel test", no sandbox. We're going to generate the
-# .gcda inplace in the bazel-out/ directory. This is in contrast to the "bazel
-# coverage" method, which is currently broken for C++ (see
-# https://github.com/bazelbuild/bazel/issues/1118). This works today as we have
-# a single coverage test binary and do not require the "bazel coverage" support
-# for collecting multiple traces and glueing them together.
-"${BAZEL_COVERAGE}" test //test/coverage:coverage_tests ${BAZEL_BUILD_OPTIONS} \
-  --cache_test_results=no --cxxopt="--coverage" --cxxopt="-DENVOY_CONFIG_COVERAGE=1" \
-  --linkopt="--coverage" --define ENVOY_CONFIG_COVERAGE=1 --test_output=streamed \
-  --strategy=Genrule=standalone --spawn_strategy=standalone --test_timeout=4000 \
-  --test_arg="--log-path /dev/null" --test_arg="-l trace"
-
-# The Bazel build has a lot of whack in it, in particular generated files, headers from external
-# deps, etc. So, we exclude this from gcov to avoid false reporting of these files in the html and
-# stats. The #foo# pattern is because gcov produces files such as
-# bazel-out#local-fastbuild#bin#external#spdlog_git#_virtual_includes#spdlog#spdlog#details#pattern_formatter_impl.h.gcov.
-# To find these while modifying this regex, perform a gcov run with -k set.
-[[ -z "${GCOVR_EXCLUDE_REGEX}" ]] && GCOVR_EXCLUDE_REGEX=".*pb\\..*|test#.*|.*#test#.*|external#.*|.*#external#.*|.*#prebuilt#.*|.*#config_validation#.*|.*#chromium_url#.*"
-[[ -z "${GCOVR_EXCLUDE_DIR}" ]] && GCOVR_EXCLUDE_DIR=".*/external/.*"
+BAZEL_USE_LLVM_NATIVE_COVERAGE=1 GCOV=llvm-profdata bazel coverage ${BAZEL_BUILD_OPTIONS} \
+    -c fastbuild --copt=-DNDEBUG --instrumentation_filter=//source/...,//include/... \
+    --test_timeout=2000 --cxxopt="-DENVOY_CONFIG_COVERAGE=1" --test_output=streamed \
+    --test_arg="--log-path /dev/null" --test_arg="-l trace" --test_env=HEAPCHECK= \
+    --test_filter='-QuicPlatformTest.QuicStackTraceTest' //test/coverage:coverage_tests
```

Comment:
Why is this test filtered? Can you add a comment or TODO?

Reply:
actually this worked in single binary so just removed filter, let's see if CI is happy with this.